store/src/java/com/zimbra/cs/html/DefangFilter.java

```diff
@@ -466,29 +467,49 @@ public void startDocument(XMLLocator locator, String encoding,
         if (value.length() < DebugConfig.defangStyleValueLimit) {
-            // remove comments
-            value = STYLE_UNWANTED_STRG_PATTERN.matcher(value).replaceAll("");
-            value = COMMENT.matcher(value).replaceAll("");
-            // strip off unwanted functions
-            Matcher matcher = STYLE_UNWANTED_FUNC.matcher(value);
-            StringBuffer stringBuffer = new StringBuffer();
-            while (matcher.find()) {
-                String match = matcher.group();
-                if (!match.startsWith("rgb") && !match.startsWith("media")
-                    && !match.startsWith("and")) {
-                    matcher.appendReplacement(stringBuffer, "");
+            int endIndex = 0;
+            int random = 0;
+            SecureRandom r = new SecureRandom();
+            int range = 500;
+            random = r.nextInt(range) + range;
+            for (int startIndex = 0; endIndex < value.length();) {
+                endIndex = startIndex + random;
+                String valuePart;
+                if (endIndex < value.length()) {
+                    valuePart = value.substring(startIndex, endIndex);
+                } else if (startIndex < value.length()) {
+                    valuePart = value.substring(startIndex);
+                } else {
+                    break;
                 }
+                // strip off unwanted functions
+                Matcher matcher = STYLE_UNWANTED_FUNC.matcher(valuePart);
+                StringBuffer stringBuffer = new StringBuffer();
+                while (matcher.find()) {
+                    String match = matcher.group();
+                    if (!match.startsWith("rgb") && !match.startsWith("media")
+                        && !match.startsWith("and")) {
+                        matcher.appendReplacement(stringBuffer, "");
+                    }
+                }
+                matcher.appendTail(stringBuffer);
+                valuePart = stringBuffer.toString();
+                sanitizedValue = sanitizedValue + valuePart;
+                random = r.nextInt(range) + range;
+                startIndex = endIndex;
             }
-            matcher.appendTail(stringBuffer);
-            value = stringBuffer.toString();
+            // remove comments
+            sanitizedValue = STYLE_UNWANTED_STRG_PATTERN.matcher(sanitizedValue).replaceAll("");
+            sanitizedValue = COMMENT.matcher(sanitizedValue).replaceAll("");
             // strip off any @import
-            ZimbraLog.mailbox.info("sanitizeStyleValue END");
-            value = STYLE_UNWANTED_IMPORT.matcher(value).replaceAll("");
+            sanitizedValue = STYLE_UNWANTED_IMPORT.matcher(sanitizedValue).replaceAll("");
         } else {
-            ZimbraLog.mailbox.info("style value is too long:%d characters.", value.length());
+            ZimbraLog.mailbox.info("style value is too long:%d characters. Removing it.", value.length());
+            sanitizedValue = "";
```

Comment:
instead of empty string, sanitised value after trimming the string should be sent.